src/test/java/com/google/sps/PersistentUserRepositoryTest.java

```diff
@@ -0,0 +1,226 @@
+    Assert.assertEquals(expected, result);
+  }
+
+  @Test
+  public void removeUserThatExists() {
+    PersistentUserRepository emptyRepo = new PersistentUserRepository();
+    emptyRepo.addUser(USER_A);
+    try {
+      emptyRepo.removeUser(USER_A);
+      String expected = "";
+      String result = emptyRepo.toString();
+      Assert.assertEquals(expected, result);
+    } catch (Exception e) {
+      Assert.fail("Exception should not be thrown in removeUserThatExists");
+    }
+  }
+
+  @Test
+  public void removeUserThatDoesNotExist() {
+    PersistentUserRepository emptyRepo = new PersistentUserRepository();
+    try {
+      emptyRepo.removeUser(USER_A);
+      Assert.fail("Exception should be thrown in removeUserThatDoesNotExist()");
+    } catch (Exception e) {
+      // don't need to do anything here because test should catch exception
+    }
+  }
+
+  @Test
+  public void getAllUsersTest() {
+      User userA = new User("John");
+      User userB = new User("Bob");
+      User userC = new User("Haley");
+
+      PersistentUserRepository myRepo = new PersistentUserRepository();
+      myRepo.addUser(userA);
+      myRepo.addUser(userB);
+      myRepo.addUser(userC);
+
+      Collection<User> results = myRepo.getAllUsers();
+      int resultsLength = results.size();
+
+      Collection<User> expected = new ArrayList<User>();
+      expected.add(userA);
+      expected.add(userB);
+      expected.add(userC);
+      int expectedLength = expected.size();
+
+      Assert.assertEquals(resultsLength, expectedLength);
+  }
+
+  @Test
+  public void addSameUserMultipleTimes() {
+      User userA = new User("John");
+
+      PersistentUserRepository myRepo = new PersistentUserRepository();
+      myRepo.addUser(userA);
+      myRepo.addUser(userA);
+      myRepo.addUser(userA);
+
+      Collection<User> allUsers = myRepo.getAllUsers();
+      int actualSize = allUsers.size();
+
+      //myRepo should only have 1 user stored
+      Assert.assertEquals(1, actualSize);
+  }
+
+  @Test
+  public void multipleAddAndRemove() {
+      User userA = new User("John");
+      User userB = new User("Bob");
+
+      PersistentUserRepository myRepo = new PersistentUserRepository();
+      myRepo.addUser(userA);
+      myRepo.addUser(userB);
+
+      Collection<User> allUsers = myRepo.getAllUsers();
+      int currentSize = allUsers.size();
+
+
+      //myRepo should have 2 users stored
+      Assert.assertEquals(2, currentSize);
+
+      try {
+        myRepo.removeUser(userB);
+      } catch (Exception e) {
+          Assert.fail("Can't remove user that does not exist");
+      }
+      allUsers = myRepo.getAllUsers();
+      //myRepo should now only have 1 user stored
+      
+      currentSize = allUsers.size();
+      Assert.assertEquals(1, currentSize);
+
+      myRepo.addUser(userB);
+      allUsers = myRepo.getAllUsers();
+      currentSize = allUsers.size();
+      //myRepo should have 2 users stored again
+      Assert.assertEquals(2, currentSize);
```

Comment:
nit: Somewhat a personal style issue, so no action required, but I'd argue inlining some of these calls makes the code much easier to read -- less code is often easer to read than more code. For instance, L188-191 could be collapsed into:

Assert.assertEquals(2, myRepo.getAllUsers().size());

Which I'd argue also removes the necessity for the comment (since you can see from the inlined version that you're checking that the number of users in myRepo is 2.

This appears throughout, no need to go back and change everything (especially if you're more inclined to explicitly name things), but something to keep in mind.

Reply:
Done